### DISCUSSION TITLE
Document supported hardware architectures

### DISCUSSION THREAD

**Discussion by zamazan4ik** [prometheus/pushgateway#502] [Q&A]
## Feature request
**Use case. Why is this important?**

I have checked the documentation but did not find any word about supported architectures.

Could you please put the information about supported architectures to the documentation please? E.e. about supported architectures for different operating systems, some specific requirements to the supported instruction sets, if you have any (e.g. maybe AVX is required - I do not know). If you support only OS-arch which are officially released - please write it somewhere in the documentation explicitly.

This kind of information is important for the end-users during the evaluation, how to install Pushgateways on their hardware.

Thanks in advance!

**Comment by beorn7** [comment]
It's a bit complicated. tl;dr: It should work everywhere where Go compiles, but it's not guaranteed.

In more detail: The Pushgateway is a fairly simple program without any special OS or hardware dependent tricks. So I would assume that any platform the Go compilare compiles for will work. _However_ that's different from actively supporting all those platforms. If a PR introduces something that is (perhaps unknowingly) OS or hardware dependent, we wouldn't really notice because we don't have test pipelines for different platforms and such. Even by compiling the binary release for different platforms, we don't really know if everything will work as expected.

If you feel an explanation along the lines of the above would be helpful, how about creating a PR against the README.md?

**Comment by beorn7** [comment]
I'll convert this to a discussion so it is available there as a source of information.
